Automattic-Tracks-iOS/Event Logging/EventLoggingDelegate.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+public protocol EventLoggingDelegate {
```

Comment:
The [WP Swift Style Guide](https://github.com/wordpress-mobile/swift-style-guide) says that we try to follow the  [raywenderlich.com Swift Style Guide](https://github.com/raywenderlich/swift-style-guide/) and that style guide mentions this for [naming delegate methods](https://github.com/raywenderlich/swift-style-guide/#delegates): 

> When creating custom delegate methods, an unnamed first parameter should be the delegate source. (UIKit contains numerous examples of this.)

This is also the standard that iOS SDK follows as far as I can tell. 

But that's up to you if you'd like to follow that. I just find that that naming pattern makes it easier to understand where methods come from and how they are used.

Reply:
This is generally a good practice for sure. In this case, I'd like to try avoiding adding `EventLoggingUploadManager` in the callback, because if a developer were to receive it, there are no public methods or properties they could work with – having the object wouldn't really help them.

Additionally, I plan to wrap `EventLoggingUploadManager` and `EventLoggingUploadQueue` in a service object that allows you to initialize all of this without being concerned with the internals – from a developer's perspective, the delegate will appear to belong to that wrapper object.

I think this'll make sense in the context of the final PR, but I'll be sure to leave a note to take another look at this as part of that one.

Reply:
👍